base/sparse/sparsevector.jl

```diff
@@ -1681,3 +1681,19 @@ droptol!(x::SparseVector, tol, trim::Bool = true) = fkeep!(x, (i, x, tol) -> abs
 
 dropzeros!(x::SparseVector, trim::Bool = true) = fkeep!(x, (i, x, other) -> x != 0, nothing, trim)
 dropzeros(x::SparseVector, trim::Bool = true) = dropzeros!(copy(x), trim)
+
+import Base.fill!
+function fill!(A:: Union{SparseVector, SparseMatrixCSC}, x)
+    # in case of fillvalue 0, only want to iterate over
+    # nonzero slots of A, thats why we have this method
+    T = eltype(A)
+    xT = convert(T, x)
+    if xT == zero(T)
+        fill!(A.nzval, xT)
+    else
+        for i in eachindex(A)
+            @inbounds A[i] = xT
```

Comment:
it's not exactly complicated to do correctly, so there's no reason to write code that will be orders of magnitude slower in a method specialized for sparse matrices